### PR TITLE
[RPA-904] Add Get cells action

### PR DESCRIPTION
**Pull request by star-in-the-jar**
## Description

returns array of arrays (where each row is an arrray - just like in the RPA framework)
image

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

**Review by star-in-the-jar — COMMENTED** [review]
self cr

**Review by star-in-the-jar — COMMENTED** [review]
self cr

File: runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts [hunk 1]
```diff
@@ -91,6 +97,27 @@ export default class ExcelActionHandler extends StatefulActionHandler {
                 .Value();
     }
 
+    async getCells(
+        input: ExcelGetCellsActionInput
+    ): Promise<Record<string, any>> {
+        const optionalWorksheet = input?.worksheet;
+        const openedWorksheet = this.session.ActiveSheet.name;
+        const cellValues: Record<string, any> = {};
```
Comment (star-in-the-jar) [inline runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts]: any to unknow

Comment (star-in-the-jar) [inline runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts]: any to unknow

**Review by star-in-the-jar — COMMENTED** [review]
self cr

File: runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts (hunk 1 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts]: and in translations change obj format to JSON obj 

+ handle wrong input -> add  

 private checkIsObject(value: unknown): void {
        if (typeof value !== 'object') {
            throw new Error(`Target Excel structure must be variable of a JSON object e.g.:
            {
                "A1": "value",
                "B3": "another value"
            }
            Check targetExcelStructure in Input tab above.`);
        }
    }

**Review by star-in-the-jar — COMMENTED** [review]
self cr

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/process/actions/excel.ts [hunk 2]
```diff
@@ -6,6 +6,8 @@ const excelActionsTranslations = {
     'Process.Details.Modeler.Actions.Excel.Open.Path.Title': 'Path / Workbook',
     'Process.Details.Modeler.Actions.Excel.Open.Worksheet.Title': 'Worksheet',
     'Process.Details.Modeler.Actions.Excel.GetCell.Label': 'Get cell',
+    'Process.Details.Modeler.Actions.Excel.GetCells.Label': 'Get cells',
+    'Process.Details.Modeler.Actions.Excel.CellCoordinates': 'An object or a list (e.g. { "A1": null, "C2": "" } or ["A1", "C2"])',
```
Comment (Klau-Sam) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/process/actions/excel.ts]: I think that if we want to get, for example, 100 cells next to each other (vertical or horizontal/both), we should be able to indicate their range without listing each one separately

**Review by Klau-Sam — CHANGES_REQUESTED** [review]

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/process/actions/excel.ts (hunk 2 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/process/actions/excel.ts]: sure
done

Comment (Klau-Sam) [reply runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/process/actions/excel.ts]: cool 🌟

**Review by star-in-the-jar — COMMENTED** [review]
self review

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/process/actions/excel.ts [hunk 3]
```diff
@@ -5,11 +5,15 @@ const excelActionsTranslations: typeof englishExcelActionsTranslations = {
...
     'Process.Details.Modeler.Actions.Excel.StartColumn': 'Kolumna początkowa',
+    'Process.Details.Modeler.Actions.Excel.EndRow': 'Wiersz końcowy',
+    'Process.Details.Modeler.Actions.Excel.EndColumn': 'Kolumna końcowa',
     'Process.Details.Modeler.Actions.Excel.Worksheet': 'Arkusz',
     'Process.Details.Modeler.Actions.Excel.Open.Label': 'Otwórz',
     'Process.Details.Modeler.Actions.Excel.Open.Path.Title': 'Ścieżka / Skoroszyt',
     'Process.Details.Modeler.Actions.Excel.Open.Worksheet.Title': 'Arkusz',
     'Process.Details.Modeler.Actions.Excel.GetCell.Label': 'Odczytaj komórkę',
+    'Process.Details.Modeler.Actions.Excel.GetCells.Label': 'Odczytaj komórki',
+    'Process.Details.Modeler.Actions.Excel.GetCells.CellCoordinates': 'Przedział lub lista np. "A1:G8" lub ["A1", "B3"]',
```
Comment (Klau-Sam) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/process/actions/excel.ts]: I don't think it's used anywhere

**Review by Klau-Sam — APPROVED** [review]

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/process/actions/excel.ts (hunk 3 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/process/actions/excel.ts]: true

Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/process/actions/excel.ts]: removed

**Pull request merged**